service/controller/v17/encrypter/kms/kms.go

```diff
@@ -44,72 +45,146 @@ func NewEncrypter(c *EncrypterConfig) (*Encrypter, error) {
 	return kms, nil
 }
 
```

Comment:
This is idempotent now. We don't have desired, current state functions which were used outside and it was confusing. The direction is to make it a `controller.Resource`. Also diagnostic logs added.